tests/lib/rules/utils/ast-utils.js

```diff
@@ -1099,7 +1099,10 @@ describe("ast-utils", () => {
             "(1, 2, foo)": true,
             "1 && 2": false,
             "1 && foo": true,
-            "foo && 2": true,
+            "foo && 2": false,
+
+            // A future improvement could detect the left side as statically falsy, making this false.
+            "false && foo": true,
```

Comment:
:+1: 

We are already doing that in `no-constant-condition` for some basic cases [here](https://github.com/eslint/eslint/blob/b5e011c865e95d700d29cb9a4ba71c671d99e423/lib/rules/no-constant-condition.js#L66), and it's going to be improved by #13245.